File: config/profiles.py
# config/profiles.py
# VoidSend - SMTP Provider Profiles

from dataclasses import dataclass
from typing import Optional

@dataclass
class SMTPProfile:
    name: str
    host: str
    port: int
    use_tls: bool
    use_starttls: bool
    free_tier_note: str
    signup_url: str
    recommended_delay: float
    max_connections: int

PROFILES: dict[str, SMTPProfile] = {
    "brevo": SMTPProfile(
        name="Brevo (formerly Sendinblue)",
        host="smtp-relay.brevo.com",
        port=587,
        use_tls=False,
        use_starttls=True,
        free_tier_note="300 emails/day free",
        signup_url="https://app.brevo.com/account/register",
        recommended_delay=0.5,
        max_connections=3,
    ),
    "mailersend": SMTPProfile(
        name="MailerSend",
        host="smtp.mailersend.net",
        port=587,
        use_tls=False,
        use_starttls=True,
        free_tier_note="3,000 emails/month free",
        signup_url="https://app.mailersend.com/register",
        recommended_delay=0.3,
        max_connections=5,
    ),
    "sendpulse": SMTPProfile(
        name="SendPulse",
        host="smtp-pulse.com",
        port=587,
        use_tls=False,
        use_starttls=True,
        free_tier_note="15,000 emails/month free",
        signup_url="https://login.sendpulse.com/registration/",
        recommended_delay=0.3,
        max_connections=5,
    ),
    "smtp2go": SMTPProfile(
        name="SMTP2GO",
        host="mail.smtp2go.com",
        port=587,
        use_tls=False,
        use_starttls=True,
        free_tier_note="1,000 emails/month free",
        signup_url="https://www.smtp2go.com/signup/",
        recommended_delay=0.5,
        max_connections=3,
    ),
    "mailgun": SMTPProfile(
        name="Mailgun",
        host="smtp.mailgun.org",
        port=587,
        use_tls=False,
        use_starttls=True,
        free_tier_note="5,000 emails/month (3-month trial)",
        signup_url="https://signup.mailgun.com/new/signup",
        recommended_delay=0.2,
        max_connections=8,
    ),
    "sendgrid": SMTPProfile(
        name="SendGrid",
        host="smtp.sendgrid.net",
        port=587,
        use_tls=False,
        use_starttls=True,
        free_tier_note="100 emails/day free forever",
        signup_url="https://signup.sendgrid.com/",
        recommended_delay=0.3,
        max_connections=5,
    ),
    "custom": SMTPProfile(
        name="Custom SMTP Server",
        host="",
        port=587,
        use_tls=False,
        use_starttls=True,
        free_tier_note="Your own SMTP host",
        signup_url="",
        recommended_delay=0.1,
        max_connections=10,
    ),
}

def get_profile(key: str) -> Optional[SMTPProfile]:
    return PROFILES.get(key.lower())

def list_profiles() -> list[tuple[str, SMTPProfile]]:
    return list(PROFILES.items())

File: src/config/profiles.rs
use serde::{Deserialize, Serialize};
use std::collections::HashMap;

/// Built-in low-cost / free SMTP profiles for 2026
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SmtpProfile {
    pub name: String,
    pub host: String,
    pub port: u16,
    pub tls: bool,           // true = STARTTLS, false = SSL/TLS
    pub username_example: String,
    pub daily_limit: Option<u32>,
    pub monthly_limit: Option<u32>,
    pub note: String,
}

pub fn get_builtin_profiles() -> HashMap<String, SmtpProfile> {
    let mut profiles = HashMap::new();

    // Brevo (formerly Sendinblue)
    profiles.insert(
        "brevo".to_string(),
        SmtpProfile {
            name: "Brevo".to_string(),
            host: "smtp-relay.brevo.com".to_string(),
            port: 587,
            tls: true,
            username_example: "your-brevo-smtp-key".to_string(),
            daily_limit: Some(300),
            monthly_limit: None,
            note: "Free tier: \\~300 emails/day. Use SMTP key as password.".to_string(),
        },
    );

    // SendPulse
    profiles.insert(
        "sendpulse".to_string(),
        SmtpProfile {
            name: "SendPulse".to_string(),
            host: "smtp-pulse.com".to_string(),
            port: 465,
            tls: false, // SSL
            username_example: "your-email@example.com".to_string(),
            daily_limit: None,
            monthly_limit: Some(12000),
            note: "Free tier: 12,000 emails/month.".to_string(),
        },
    );

    // SMTP2GO
    profiles.insert(
        "smtp2go".to_string(),
        SmtpProfile {
            name: "SMTP2GO".to_string(),
            host: "mail.smtp2go.com".to_string(),
            port: 587,
            tls: true,
            username_example: "your-smtp2go-username".to_string(),
            daily_limit: None,
            monthly_limit: Some(1000),
            note: "Free tier: 1,000 emails/month.".to_string(),
        },
    );

    // MailerSend
    profiles.insert(
        "mailersend".to_string(),
        SmtpProfile {
            name: "MailerSend".to_string(),
            host: "smtp.mailersend.net".to_string(),
            port: 587,
            tls: true,
            username_example: "MS_xxxxxxxx".to_string(),
            daily_limit: None,
            monthly_limit: Some(500),
            note: "Free tier: 500 emails/month.".to_string(),
        },
    );

    profiles
}

pub fn list_profiles() {
    let profiles = get_builtin_profiles();
    println!("VoidSend - Built-in SMTP Profiles (2026 free tiers):\n");
    for (key, p) in profiles {
        println!("• {} ({})", p.name, key);
        println!("  Host : {}:{}", p.host, p.port);
        println!("  TLS  : {}", if p.tls { "STARTTLS" } else { "SSL" });
        println!("  Limit: {} daily / {} monthly", 
            p.daily_limit.map_or("N/A".to_string(), |v| v.to_string()),
            p.monthly_limit.map_or("N/A".to_string(), |v| v.to_string()));
        println!("  Note : {}\n", p.note);
    }
}
